pages/settings.tsx

```diff
@@ -476,6 +493,13 @@ class Settings extends Component<Props, State> {
             <Fragment>
               <hr />
               <h4>{t('MyData')}</h4>
+              <p>{t('Export OPML Description')}</p>
+              <Button
+                className='settings__download'
+                onClick={this.downloadLoggedInOPMLData}>
```

Comment:
I discarded isLoading={isDownloading} for this button. Is it right?

Reply:
Ah good catch, I missed that.

The idea to provide a loading spinner in the button while the asynchronous operation is running. It should be pretty easy to add some state to make this work for the OPML Export button.

Try adding this to the State type on the page:

`isDownloadingOPML?: boolean`

Then on the first line inside of `downloadLoggedInOPMLData` add the line:

`this.setState({ isDownloadingOPML: true })`

and then on the last line of that function call:

`this.setState({ isDownloadingOPML: false })`

Then in the `render` method grab `isDownloadingOPML` off the state, and pass it into `isLoading={isDownloadingOPML}` for the Export OPML component.